lib/buildkit/client.rb

```diff
@@ -120,6 +128,34 @@ def request(method, path, data, options = {})
       response.data
     end
 
+    def extract_query_and_headers_from(data)
+      {
+        query: data.delete(:query) || {},
+        headers: data.delete(:headers) || {},
+      }
+    end
+
+    def paginate(method, path, data, options = {})
+      response = []
+      loop do
```

Comment:
Do we have to be concerned about being throttled on the requests and potentially track an exponential backoff or similar case?

Reply:
The documentation does not mention anything about request rate limit.